### PR TITLE
[stable/prestashop] Add ingress

### LINKED ISSUE #9359: [stable/prestashop] missing ingress template
**Is this a request for help?**:
No, this is feature request, in the prestashop chart there isn't an ingress 
---

**FEATURE REQUEST** :

  - Add ingress template in the prestashop chart

**Which chart**:
 stable/prestashop

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

#### What this PR does / why we need it:
Add Ingress support to the chart 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #9359 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/prestashop/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {{- end }}
+    {{- if .certManager }}
+    kubernetes.io/tls-acme: "true"
+    {{- end }}
+    {{- range $key, $value := .annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  rules:
+  - host: {{ .name }}
```
Comment (Nittarab) [inline stable/prestashop/templates/ingress.yaml]: Can we use the  
`- host: {{ template "prestashop.host" }}`

Comment (Nittarab) [inline stable/prestashop/templates/ingress.yaml:24]: Can we use the template prestashop host? 
`- host: {{ template "prestashop.host" }}`

File: stable/prestashop/values.yaml [hunk 2]
```diff
@@ -216,6 +216,52 @@ readinessProbe:
...
+## ref: http://kubernetes.io/docs/user-guide/ingress/
+##
+ingress:
+  ## Set to true to enable ingress record generation
+  enabled: false
+
+  ## The list of hostnames to be covered with this ingress record.
+  ## Most likely this will be just one host, but in the event more hosts are needed, this is an array
+  hosts:
+  - name: prestashop.local
```
Comment (Nittarab) [inline stable/prestashop/values.yaml:230]: `- name: prestashop.host`

File: stable/prestashop/values.yaml [hunk 3]
```diff
@@ -216,6 +216,52 @@ readinessProbe:
...
+
+    ## Set this to true in order to enable TLS on the ingress record
+    ## A side effect of this will be that the backend wordpress service will be connected at port 443
```
Comment (Nittarab) [inline stable/prestashop/values.yaml:233]: Remove the `A side effect` line

File: stable/prestashop/values.yaml [hunk 4]
```diff
@@ -216,6 +216,52 @@ readinessProbe:
...
+    tls: false
+
+    ## Set this to true in order to add the corresponding annotations for cert-manager
+    certManager: false
+
+    ## If TLS is set to true, you must declare what secret will store the key/certificate for TLS
+    tlsSecret: wordpress.local-tls
```
Comment (Nittarab) [inline stable/prestashop/values.yaml:240]: Change the wordpress with prestahop and include the info about create and manage the certificates outside of this helm chart

**Review by Nittarab — COMMENTED** [review]
Can we use the template `prestashop.host`? 
Maybe we can add some info about tls key management?

**Comment by Nittarab** [comment]
@javsalgar any updates?

**Comment by juan131** [comment]
@Nittarab I created https://github.com/helm/charts/pull/10335 to fix the same issue using a different approach.

**Pull request closed without merging**